src/analysis/liouville/liouville_constant.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Jujian Zhang
+-/
+import analysis.liouville.basic
+/-!
+
+# Liouville constants
+
+This file contains a construction of a family of Liouville numbers.
+The most important property is that they are examples of transcendental real numbers.
+This fact is recorded in `is_liouville.is_transcendental_of_liouville_constant` (a result in
+a subsequent PR).
+-/
+
+noncomputable theory
+open_locale nat big_operators
+open set real finset
+
+section m_is_real
+
+variable {m : ℝ}
+
+namespace liouville
+
+/--
+For a real number `m`, Liouville's constant is
+$$
+\sum_{i=0}^\infty\frac{1}{m^{i!}}.
+$$
+The series converges only for `1 < m`.  However, there is no restriction on `m`, since,
+if the series does not converge, then the sum of the series is defined to be zero.
+-/
+def liouville_number (m : ℝ) := ∑' (i : ℕ), 1 / m ^ i!
+
+/--
+`liouville_constant_first_k_terms` is the sum of the first `k` terms of Liouville's constant, i.e.
+$$
+\sum_{i=0}^k\frac{1}{m^{i!}}.
+$$
+-/
+def liouville_number_first_k_terms (m : ℝ) (k : ℕ) := ∑ i in range (k+1), 1 / m ^ i!
```

Comment:
Isn't this actually a sum of k+1 terms?
```suggestion
def liouville_number_first_k_terms (m : ℝ) (k : ℕ) : ℝ := ∑ i in range (k+1), 1 / m ^ i!
```

Reply:
I can't think of a good name suggestion `liouville_number_first_k_plus_one_terms` is a bit awkward.

Reply:
You are probably right: I simply "inherited" this from Jujian, but I will check and think of a different name once I am back at a computer!

Reply:
How about "initial terms"?

Reply:
That sounds good to me!

Reply:
Bryan, I changed to `initial` and `tail`, so there is no need to explicitly mention `k` in the name, since that is simply a variable.  Let me know if you prefer something else!

Reply:
Perfect!